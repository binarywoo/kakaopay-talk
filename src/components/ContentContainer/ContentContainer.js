import React from 'react'

const ContentContainer = ({ children, ...restProps }) => {
  return (
    <div className='content-container' {...restProps}>
      {children}
    </div>
  )
}

export default ContentContainer
