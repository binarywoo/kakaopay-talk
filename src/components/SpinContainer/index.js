import React from 'react'
import { Spin } from 'antd'

const SpinContainer = ({ height = '47vh', size, tip }) => {
  return (
    <div
      style={{ textAlign: 'center', paddingTop: height, paddingBottom: height }}
    >
      <Spin size={size} tip={tip} />
    </div>
  )
}

export default SpinContainer
