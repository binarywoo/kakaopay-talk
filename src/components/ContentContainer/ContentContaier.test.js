import React from 'react'
import { shallow } from 'enzyme'
import ContentContainer from './index'
import SpinContainer from '../SpinContainer/index'

describe('ContentContainer', () => {
  let wrapper = null

  it('정상적으로 렌더링이 된다.', () => {
    wrapper = shallow(
      <ContentContainer>
        <SpinContainer />
      </ContentContainer>
    )
  })

  it('스냅샷과 매칭이 된다.', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('자식컴포넌트를 정상적으로 랜더링 한다.', () => {
    expect(wrapper.children()).not.toBeNull()
  })
})
