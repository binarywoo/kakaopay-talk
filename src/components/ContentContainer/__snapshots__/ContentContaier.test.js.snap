// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer 스냅샷과 매칭이 된다. 1`] = `
<div
  className="content-container"
>
  <SpinContainer />
</div>
`;
